### PR TITLE
Updated openjdk_test file - use of ci.role.perf label in Jenkins 

### PR DESCRIPTION
**Pull request by norbline**

### CODE REVIEW

File: buildenv/jenkins/openjdk_tests [hunk 1]
```diff
@@ -152,7 +152,7 @@ if (JDK_VERSIONS.size() > 1 || JDK_IMPLS.size() > 1 || PLATFORMS.size() >1 || PL
 } else {
     if (PLATFORM_MAP.containsKey(params.PLATFORM)) {
         SPEC = PLATFORM_MAP[params.PLATFORM]["SPEC"]
-        LABEL = params.LABEL ? params.LABEL : PLATFORM_MAP[params.PLATFORM]["LABEL"]
+        LABEL = BUILD_LIST.contains("perf") ? params.LABEL : PLATFORM_MAP[params.PLATFORM]["LABEL"]
```
Comment (llxia) [inline buildenv/jenkins/openjdk_tests:155]: This does not set `ci.role.perf` in LABEL

**Comment by llxia** [comment]
The change of adding `ci.role.perf ` needs to wait until both internal and external Jenkins have the machine labeled properly.

I think we should start with documenting the Perf machine requirements and `ci.role.perf ` machines at Adopt Jenkins. Ideally, we need at least one perf machine on each platform.

**Comment by norbline** [comment]
> The change of adding `ci.role.perf ` needs to wait until both internal and external Jenkins have the machine labeled properly.
> 
> I think we should start with documenting the Perf machine requirements and `ci.role.perf ` machines at Adopt Jenkins. Ideally, we need at least one perf machine on each platform.

Hi @llxia looking at @sxa comment https://github.com/adoptium/aqa-tests/issues/2401#issuecomment-817853626.. is like he workes on it or something else

File: buildenv/jenkins/openjdk_tests [hunk 2]
```diff
@@ -152,8 +152,15 @@ if (JDK_VERSIONS.size() > 1 || JDK_IMPLS.size() > 1 || PLATFORMS.size() >1 || PL
...
-
+      if (params.LABEL) {
+          LABEL = params.LABEL
+} else {
+  LABEL = PLATFORM_MAP[params.PLATFORM]["LABEL"]
+  if (params.BUILD_LIST.contains("perf")) {
+     LABEL = LABEL.minus("ci.role.test&&")
+     LABEL += '&&ci.role.perf'
+  }
+}
```
Comment (smlambert) [inline buildenv/jenkins/openjdk_tests:163]: Thanks @norbline - can you format/indent the block so its formatted like the other code?

**Review by smlambert — COMMENTED** [review]
We could merge this as is, in which case if there are any platforms where we do not have machines, the test jobs would fail to find a machine that is online with a matching LABEL, https://github.com/adoptium/aqa-tests/blob/7ebdbe3d1408a93747a2b89cbb6181dddfd42cbe/buildenv/jenkins/openjdk_tests#L181-L182.  

An improvement to this (which we can do as part of this PR, or in a follow-up PR), is to move L181-182 to its own method, areNodesWithLabelOnline() that returns a boolean and call it in your else block to check we have nodes with your newly built LABEL, if not, then print out a warning and drop back to using ci.role.test for the time being.

**Review by sophia-guo — APPROVED** [review]

**Comment by smlambert** [comment]
hi @norbline - based on my chat with @llxia and since we are trying not to break anyone who is currently using our scripts, the prudent thing to do as part of this PR is to create a method that checks if nodes of a particular label are online, something like:
```
def areNodesWithLabelOnline(labelToCheck) {
            String[] onlineNodes = nodesByLabel(labelToCheck)
            if (onlineNodes.size() < 1) {
                        return false
            } 
            return true
}        
```
before changing label, in your L159 check 
```
if (params.BUILD_LIST.contains("perf") && areNodesWithLabelOnline(LABEL.minus("ci.role.test&&").concat("&&ci.role.perf")) {
```

and since we have a method, also replace [L181-L182](https://github.com/adoptium/aqa-tests/blob/7ebdbe3d1408a93747a2b89cbb6181dddfd42cbe/buildenv/jenkins/openjdk_tests#L181-L182) with 
```
areNodesWithLabelOnline(LABEL) {
```

Can you make these changes to this PR please?  then we can proceed to test and merge.  Thanks!

**Comment by norbline** [comment]
> hi @norbline - based on my chat with @llxia and since we are trying not to break anyone who is currently using our scripts, the prudent thing to do as part of this PR is to create a method that checks if nodes of a particular label are online, something like:
> 
> ```
> def areNodesWithLabelOnline(labelToCheck) {
>             String[] onlineNodes = nodesByLabel(labelToCheck)
>             if (onlineNodes.size() < 1) {
>                         return false
>             } 
>             return true
> }        
> ```
> 
> before changing label, in your L159 check
> 
> ```
> if (params.BUILD_LIST.contains("perf") && areNodesWithLabelOnline(LABEL.minus("ci.role.test&&").concat("&&ci.role.perf")) {
> ```
> 
> and since we have a method, also replace [L181-L182](https://github.com/adoptium/aqa-tests/blob/7ebdbe3d1408a93747a2b89cbb6181dddfd42cbe/buildenv/jenkins/openjdk_tests#L181-L182) with
> 
> ```
> areNodesWithLabelOnline(LABEL) {
> ```
> 
> Can you make these changes to this PR please? then we can proceed to test and merge. Thanks!

wow, thanks so much for the help @smlambert...

**Comment by smlambert** [comment]
Note: I did not check yet if my code suggestions will work, so once you make the changes, you will need to run some Grinders on the changes @norbline :)

**Comment by norbline** [comment]
> Note: I did not check yet if my code suggestions will work, so once you make the changes, you will need to run some Grinders on the changes @norbline :)
is this okay @smlambert ..failed on grinder https://ci.adoptopenjdk.net/job/Grinder/2778/
stage('Queue') {
            def areNodesWithLabelOnline(labelToCheck) {
            String[] onlineNodes = nodesByLabel(labelToCheck)
            if (onlineNodes.size() < 1) {
                        return false
            }
            return true
} {
                int ACTIVE_NODE_TIMEOUT = params.ACTIVE_NODE_TIMEOUT ? params.ACTIVE_NODE_TIMEOUT : 0
                timeout(ACTIVE_NODE_TIMEOUT) {
                    // If there is available node before timeout
                    node(LABEL) {
                        echo "find the node with label as ${env.NODE_NAME}"
                    }
                }
            }

**Comment by smlambert** [comment]
hi @norbline - I may have caused you to go in the wrong direction :)

The new method you added should not go where you have put it, but rather it can go near the end of the file just before the other method, checkErrors(errorList) is defined at Line303.

Also, the reformatting you have applied is not desirable, as it has gone and changed all of the " double quotes to ' single quotes, which is unneeded.  Perhaps you can undo your last commit and just adjust the indentation of the small section of code by hand (without invoking formatting in your IDE).

Message me if you want to connect and walk through the next code review over a video call where we share a screen.  :)

**Pull request closed without merging**